platform/pc/boot/stage1.s

```diff
@@ -75,11 +75,42 @@ dap:
         .lba          dq 1
 
 
+%ifdef DEBUG
+bits 16
+%include "debug.inc"
+bits 32
+%endif
+
 readsectors:
+        mov eax, [dap.sector_count]
```

Comment:
sector_count is two bytes, so shouldn't this use ax instead of eax? (same applies to all uses of eax / ecx below)